[Rachel Ho] iP

rachel170 opened the pull request.

Commit:
Add Gradle support

rachel170 committed:
Modified Greetings, added Echo feature and Exit feature

rachel170 committed:
Added ability to store text input from user

rachel170 committed:
Added ability to display stored user inputs

rachel170 committed:
Customized some of Duke's outputs

rachel170 committed:
Added Task Class and ability to mark Task as done

rachel170 committed:
Added Event, Deadline and Todo class inherited from Task, suppoorts tracking for 3 task types

rachel170 committed:
Added Semi-Automatic Testing

rachel170 committed:
Moditified printList to handle empty list prints

rachel170 committed:
Added DukeException class to handle error in user input

rachel170 committed:
Added function to delete tasks from list. Also modified list to be ArrayList and removed counter for list. Updated semi-automatic test cases.

rachel170 committed:
Replaces "\t" with 4 spaces

As part of customization, tab was used instead of simply spaces.

Not all computer's tab spaces are the same.

Let's replace tab spaces with 4 spaces in the strings 
to produce similar UI visuals for Duke.

rachel170 committed:
Adds Save function

Part of the project requirements.


Save the tasks in the hard disk automatically 
whenever the task list changes. 

Load the data from the hard disk when Duke starts up.

Data loaded in from "/data/duke.txt".
May not be OS independent.

Loading of data is not equipped to handle data not in
specific format.

rachel170 committed:
Adds Date recognition for Deadline Tasks

Part of Project Requirements (Level-8).

Takes in dates for Deadline within strings in the 
format of "yyyy-mm-dd" and displays it as "dd MMM yyyy".

Some code was added to Event class to prepare for
future modifications that will allow Duke to recognize
both Date and Time within the command string of an
Event Task.

rachel170 committed:
Merge branch 'branch-Level-7'

Merges branch-Level-7 to master branch.

Updates master with code updates that implements
the Save function in Duke.

rachel170 committed:
Resolves merge conflicts for branch-Level-8 to master merge

rachel170 committed:
Updates loadFileContents() method to include Level-8 additions

Save function was done without taking into account
the changes made to let Deadline tasks recognize their
string inputs as dates. Since loadFileContents() method
lets Duke load old data, the method needed to be 
updated to properly recognize dates from the saved
data file (if any).

Also updates some comments so that they are no 
longer inaccurate.

rachel170 committed:
Adds 5 classes to improve OOP and refactor the code

Part of A-MoreOOP requirements.

Refactors the code to extract out closely related code as classes.

Adds the following classes,
- Ui: deals with interactions with the user
- Storage: deals with loading tasks from the file and saving tasks in the file
- Parser: deals with making sense of the user command
- TaskList: contains the task list e.g., it has operations to add/delete tasks in the list
- Command: enum class to denote commands parsed by the Parser

Commit is untested. May require more edits before pushing.

rachel170 committed:
Fixes minor bugs from last commit

The previous commit adds 5 new classes to improve OOP and
also refactors the code a little. The commit did not undergo testing.

Several testing was done and fixes were made based off
the test results.

The following are some of the fixes made:
- Adjusted some UI element to be more uniform.
- Fixed TaskList deleteTask() and markTaskDone() methods
  to perform correct actions on the right index. Also changed
  the exception catching to catch IndexOutOfBoundsException
  instead of ArrayIndexOutOfBoundsException.

rachel170 committed:
Adds a few JUnit tests for Parser and TaskList

For project requirement A-JUnit.

Adds several JUnit test methods for Duke,
- Adds a dummy test method in DukeTest class.
- Adds a few test methods in ParserTest that checks 
  invalid command, valid Todo commands, valid and 
  invalid Deadline commands.
- Adds test methods in TaskListTest to that tests invalid index 
  error when using the 'done' command.

rachel170 committed:
Adds line break after bye command

Adjusting UI components. Modified Parser class for it.

rachel170 committed:
Fixes Storage saveToFile() method

Duke could not create file to save data in due to missing line.

Commit fixes this by adding the line with createFile()
in saveToFile() method.

rachel170 committed:
Adds JavaDoc comments to all classes

rachel170 committed:
Adds JavaDoc commentary to all class methods

rachel170 committed:
Adds JavaDoc comments to commit classes

rachel170 committed:
Modifies code to adhere to Coding Standard

rachel170 committed:
Adds Find function to Duke

Project requirements for Level-9.

Method added to TaskList class.
Adjustments added to Command, Duke, Parser to accommodate 
new command.

rachel170 committed:
Resolves Merge Conflict

For merge from branch-A-CodingStandard to master.

rachel170 committed:
Resolves merge conflict

For merge of branch-Level-9 to master.

rachel170 committed:
Updates Level-9 to changes made during merges

The searchForKeyword() method uses the old variable name,
which is now changed to tasks in A-CodingStandard.

Changes has been made to fix this.

rachel170 committed:
Removes runtest.sh in text-ui-test folder

rachel170 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Wincenttjoi commented on build.gradle:
```diff
@@ -0,0 +1,46 @@
+}
+
+application {
+    mainClassName = "seedu.duke.Duke"
```
Does your gradle able to handle building/ running your application? You can change the mainClassName into your application's class name if it does not work.

Wincenttjoi commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,17 @@
+ * Types of task that can be made:
+ * DEADLINE, EVENT, TO DO
+ * */
+public enum Command {
```
I like how you put the Commands into ENUM class for checking :) Great job!

Wincenttjoi commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,37 @@
+
+/**
+ * Deadline inherits from Task.
+ *
```
Is there any reason why there is an empty line here? For this module, I think the separation of line is between description and @params. Perhaps you can consider using <p> to separate the description into different paragraphs :D

Wincenttjoi replied:
I notice that the same issue in other places too. Feel free to edit!

Wincenttjoi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,169 @@
+     */
+    public static Command parse(String userInput) throws DukeException {
+        // Check if it is a single word command
+        if(userInput.equals("bye")) {
```
Should there be spacing here: if[space](userInput ?

rachel170 replied:
Thanks for notifying me about this!

Wincenttjoi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,169 @@
+                        + instructions[0] + " command. Sorry! D:");
+            }
+        } else if (instructions[0].equals("deadline")) {
+            /** Command: "deadline <taskName> /by <deadline>" **/
```
Is there any reason why you use /** instead of // ? From my understanding, /** is used for JavaDoc and // for specific comments. We can check with prof regarding this!

Wincenttjoi commented on src/test/java/ParserTest.java:
```diff
@@ -0,0 +1,77 @@
+            actual = e.getMessage();
+        }
+        String expected = "todo read book";
+        assertEquals(expected, actual);
```
I like how you organized the different unit tests into its respective classes. However, is there a reason why you use assertEquals rather than assertThrows? Its not a major issue but something to consider as it will make your code neater

Wincenttjoi commented on src/main/java/Ui.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    /**
+     * Print's Duke's farewell to user.
```
Just a minor spelling issue, can consider changing print's into prints :)

Wincenttjoi commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,176 @@
+
+    /**
+     * Searches through all of Duke's tasks and
+     * print out tasks with given keyword
```
Should this be prints?

Wincenttjoi reviewed (commented):
Overall LGTM! Just a few nits to change such as minor violations in JavaDoc and comments section.
Other than that, would be good if you can consider using root word for the git commit verbs moving forward. Eg: **Add** instead of **Adds** - as this is the git standard that we follow :)

rachel170 committed:
Updates Duke to use packages

Duke has all its classes in one folder.

Changes made:
- Moves classes related to Task into util/task folder.
- Moves every other class except for Duke into util folder.
- Updates the class with packages and imports them into Duke.

rachel170 committed:
Updates build.gradle mainClassName

Directs gradle to Duke based off own project layout.

rachel170 committed:
Updates classes to Gradle implementation

Changes made,
- Edits some JavaDoc comment to ensure it complies to the standards.
- Removes emote from Storage class due to issues.
- Updates JUnit tests to import packages.

rachel170 committed:
Merge branch 'branch-A-Gradle'
Updates master branch with updates for Gradle.

rachel170 committed:
Merge branch 'branch-A-Gradle' into branch-Level-10
Updates branch-LEvel-10 with Gradle changes made in branch-A-Gradle.

davidliew9 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,171 @@
+     */
+    public static void checkDate(String dateString) throws DukeException {
+        try {
+            LocalDate df = LocalDate.parse(dateString);
```
Perhaps could use a better variable name rather than 'df'

danadi7 replied:
I think this is not an issue as this is a scratch variable used only on this line.

davidliew9 commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,155 @@
+     */
+    public ArrayList<Task> loadFileContents() throws FileNotFoundException {
+        // Initialize list to be returned
+        ArrayList<Task> list = new ArrayList<>(100);
```
naming of list could be more specific?

danadi7 replied:
Agree with this. Perhaps consider renaming it to something more specific?

davidliew9 commented on src/main/java/util/task/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+public class Deadline extends Task {
+    protected LocalDate dueDate;
+
+    public Deadline(String name, LocalDate details) {
```
perhaps a more specific name rather than 'details' such as 'deadline' or 'duedate'

danadi7 replied:
Agree with this. `details` does not identify it as a LocalDate object, though as this is just a temporary variable it is not a big issue.

davidliew9 reviewed (commented):
Overall well documented and easy to follow with good naming conventions. Only minor violations with regards to variable names

rachel170 committed:
Adds JavaFX implementation for Duke GUI

Basic implementation only, Duke is not actually usable
through the implementation.

rachel170 committed:
Udates Duke and other classes to return String

Duke and other classes made use of void methods in
which System.out.println() was used to print messages
to display to users. 

Changes made,
- Creation of task, marking task done, deletion of task and
  finding tasks with keyword methods changed to methods
  that return String.

- UI class was updated to reflect the changes done in TaskList and Duke.

- Duke was updated so that UI will print command messages
  passed into them from TaskList.

- Task class updated to remove tick and cross for status
  due to showing up as indecipherable symbols in GUI.

- MainWindow classes adjusted to print Duke's greeting message.

- Main class updated to display window title in GUI.

rachel170 committed:
Udates Duke and other classes to return String

Duke and other classes made use of void methods in
which System.out.println() was used to print messages
to display to users. 

Changes made,
- Creation of task, marking task done, deletion of task and
  finding tasks with keyword methods changed to methods
  that return String.
- UI class was updated to reflect the changes done in TaskList and Duke.
- Duke was updated so that UI will print command messages
  passed into them from TaskList.
- Task class updated to remove tick and cross for status
  due to showing up as indecipherable symbols in GUI.
- MainWindow classes adjusted to print Duke's greeting message.
- Main class updated to display window title in GUI.

rachel170 committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

rachel170 committed:
Removes Duke:run() and methods related to it

Minor change done to UI:getList() to fix display.

rachel170 committed:
Merge branch 'branch-Level-10'
Updates master branch with code to give Duke a GUI
For iProject Level-10 Requirements.
Code for old console run Duke removed as well.

danadi7 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,101 @@
+            // Use switch to process the commands
+            switch (cmd) {
+            case EXIT:
+                String error = "\n";
+                // Try to Save the data
+                try {
+                    storage.saveToFile(tasks.getTasks());
+                } catch (DukeException e) {
+                    error = ui.getError(e);
+                } finally {
+                    // Return farewells
+                   return ui.getFarewells() + error;
+                }
```
Naming a new line as `error` and then overriding it with an actual error may be confusing. Perhaps consider making separate variables for each.

danadi7 commented on src/main/java/Main.java:
```diff
@@ -0,0 +1,30 @@
+            e.printStackTrace();
+        }
+    }
+}
```
Not explicitly in the coding standards, but I think it's traditional / standard practice to end a file with a blank new line, as highlighted by GitHub here.

danadi7 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,171 @@
+    }
+
+    /**
+     * Returns lateral location of the specified position.
+     * If the position is unset, NaN is returned.
```
Javadoc not descriptive of function! I think you copy-pasted and forgot to change here. 😝

danadi7 commented on src/main/java/util/Parser.java:
```diff
@@ -0,0 +1,171 @@
+     * @param dateString String that may be a LocalDate.
+     * @throws DukeException  If dateString is not a LocalDate.
+     */
+    public static void checkDate(String dateString) throws DukeException {
+        try {
+            LocalDate df = LocalDate.parse(dateString);
+        } catch (Exception e) {
+            // Check if string can be recognized as a valid LocalDate
+            // If can't, print out error message
+            throw new DukeException("Whoops! I think there is an error in your date." +
+                    "\nPlease Try Again!");
+        }
+    }
+}
```
Seems like the purpose of this function is to catch an exception and throw a different exception in its place. Not sure if this is good practice as it may mask errors.

danadi7 commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,125 @@
+public class Ui {
+    /** Constants **/
+    private final String outputBreaker = ">>> ";
+    private final String lineBreaker = "--.--.--.--.--.--.--.--.--.--.--." +
```
Perhaps consider naming it just `lineBreak`? Naming it `lineBreaker` makes it seem like it has a function.

rachel170 replied:
That is a good point, thanks!

danadi7 commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,155 @@
+        ArrayList<Task> list = new ArrayList<>(100);
+
+        File data = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(data); // create a Scanner using the File as the source
```
You named `Scanner` as `scanner` in some classes, and `s` in others. Do consider keeping the names consistent for ease of reading the code.

rachel170 replied:
Noted!

danadi7 reviewed (commented):
Overall, I would say it is well written, and conforms to almost all the coding standards. Does things somewhat differently from the other PR I reviewed, and I learnt some new stuff in the process.

Only issue that caught my eye was variable naming: some variable names were not really descriptive, and some were inconsistent. Besides that, looks good to me!

rachel170 committed:
Adds Java assert to Duke and Storage class

Part of iProject requirements A-Assertions.

Use assert feature (not JUnit assertions) to document 
important assumptions that should hold at various 
points in the code.

rachel170 committed:
Corrects an assertion in Storage class

rachel170 committed:
Corrects several code quality issues

Covers danadi7's PR review comments on github.

rachel170 committed:
Update code quality based off PR review

Based off Wincenttjoi comments in the PR review.

rachel170 committed:
Remove dead code

Found several dead code from before Duke's GUI
implementation. 

Deleted the dead code.

rachel170 committed:
Merge pull request #1 from rachel170/branch-A-Assertions

Add Java assert to parts of code

rachel170 committed:
Update branch-A-CodeQuality with master

Attempted merge from master into branch-A-CodeQuality.

Failure due to recent merge from branch-A-Assertions to master.

Settle merge conflict found in Storage class.

rachel170 committed:
Storage: changed list to tasks

Merge conflict resolution kept some tasks variable as 
the list, which is the old name of the variable.

rachel170 committed:
Merge pull request #2 from rachel170/branch-A-CodeQuality

Update code quality

rachel170 committed:
Merge branch 'master' into branch-C-Update

Updates branch-C-Update with merges from 
A-Assertions and A-CodeQuality changes.

rachel170 committed:
Add Update Task Description function

iProject requirements for C-Update.

Give Duke the ability to change the description of any
task in its TaskList.

rachel170 committed:
Merge branch 'branch-C-Update'
Add update task description function to Duke

rachel170 committed:
Add Checkstyle to Duke

Checkstyle from address-book-lvl3.

Checkstyle violations fixed.

rachel170 committed:
Remove DukeTest.java

DukeTest class only contains dummy test case.

rachel170 committed:
Edit Duke's GUI

Changes color scheme and adds padding to text.

rachel170 committed:
Extend Duke's save data feature to closing app

rachel170 committed:
Change error message to show deadline's dead format

rachel170 committed:
Add focus on keyword that cannot be found

rachel170 committed:
Update Duke's GUI

rachel170 committed:
Add User Guide for Duke

rachel170 committed:
Change profile image to .png format

rachel170 committed:
Return user guide images for release

rachel170 committed:
Delete Ui.PNG

Incorret name

rachel170 committed:
Add Ui.png to docs

rachel170 committed:
Update image sizing for example in user guide

rachel170 committed:
Update Ui.png cropping

rachel170 committed:
Update UG

Fix image addressing

rachel170 committed:
Set theme jekyll-theme-slate

rachel170 committed:
Fix UG

rachel170 committed:
Fix UG images

rachel170 committed:
Fix image file path for UG

rachel170 committed:
Fix UG: examples
